store/plugins/preferencesSyncPlugin.ts

```diff
@@ -1,12 +1,91 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
 import { Plugin } from 'vuex';
-import { AppState } from '..';
+import { AppState } from '~/store';
+import { syncedMutations } from '~/store/clientSettings';
 
 export const preferencesSync: Plugin<AppState> = (store) => {
-  store.subscribe((mutation) => {
-    if (mutation.type.includes('settings/')) {
-      store.dispatch('displayPreferencesApi/updateSettings', null, {
-        root: true
-      });
+  store.subscribe(async (mutation) => {
+    if (
+      syncedMutations.some((syncedMutation) =>
+        mutation.type.includes(syncedMutation)
+      ) &&
+      store.$auth.loggedIn
+    ) {
+      try {
+        await store.dispatch('setSyncStatus', true);
+        // We set a new last sync date at the start, so if the push fails, we still have the last attempt's
+        // date and we can compare with the server when we're back online
+        await store.dispatch('clientSettings/setLastSyncDate');
```

Comment:
This should be in the root one so it's independent on whatever store we have. 

If one push fails, I think we should just simply pull/push all of the different settings we have, I don't see a real benefit of handling them independently, it's just more complication.

Reply:
As mentioned before, we don't want to sync everything at the same time. It's nicer for organization to have settings split into multiple stores, depending on what they mess with. Syncing all of these at once would generate multiple requests.

We also want to be able to keep track of individual store's sync date, so we know when it was last modified on the server vs on our side.